apollo-router/src/services/execution_service.rs

```diff
@@ -174,21 +179,74 @@ impl ExecutionService {
 
         let execution_span = Span::current();
 
-        let stream = stream
-            .filter_map(move |response: Response| {
-                ready(execution_span.in_scope(|| {
-                    Self::process_graphql_response(
-                        &query,
-                        operation_name.as_deref(),
-                        &variables,
-                        is_deferred,
-                        &schema,
-                        &mut nullified_paths,
-                        response,
-                    )
-                }))
-            })
-            .boxed();
+        let stream =
+            stream
+                .map(move |mut response: Response| {
+                    // Enforce JWT expiry for deferred and subscription responses
```

Comment:
could it be enforced on the request side, before executing the subscription?

Reply:
Oh yes good catch ! I think you can move your logic right [here](https://github.com/apollographql/router/blob/dev/apollo-router/src/services/supergraph_service.rs#L515) gary to not try to fetch data from subgraphs if your jwt is expired

Reply:
I've split the logic for enforcement and now enforce subscription expiration in the request as @bnjjj suggested. I've left the defer enforcement in the response. We could remove that I suppose, but it's probably better to have it than not.